fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/RuntimeFilterGenerator.java

```diff
@@ -131,6 +137,41 @@ public PhysicalPlan visitPhysicalHashJoin(PhysicalHashJoin<? extends Plan, ? ext
         return join;
     }
 
+    /**
+     * consider L join R on L.a=R.b
+     * runtime-filter: L.a<-R.b is effective,
+     * if R.b.selectivity<1 or b is partly covered by a
+     *
+     * TODO: min-max
+     * @param equalTo join condition
+     * @param join join node
+     * @return true if runtime-filter is effective
+     */
+    private boolean isEffectiveRuntimeFilter(EqualTo equalTo, PhysicalHashJoin join) {
+        StatsDeriveResult leftStats = ((AbstractPlan) join.child(0)).getStats();
+        StatsDeriveResult rightStats = ((AbstractPlan) join.child(1)).getStats();
+        Set<Slot> leftSlots = equalTo.child(0).getInputSlots();
+        if (leftSlots.size() > 1) {
+            return false;
+        }
+        Set<Slot> rightSlots = equalTo.child(1).getInputSlots();
+        if (rightSlots.size() > 1) {
+            return false;
+        }
+        Slot leftSlot = leftSlots.iterator().next();
+        Slot rightSlot = rightSlots.iterator().next();
+        ColumnStat probeColumnStat = leftStats.getColumnStatsBySlot(leftSlot);
+        ColumnStat buildColumnStat = rightStats.getColumnStatsBySlot(rightSlot);
+        if (probeColumnStat == null || buildColumnStat == null) {
+            probeColumnStat = leftStats.getColumnStatsBySlot(rightSlot);
+            buildColumnStat = rightStats.getColumnStatsBySlot(leftSlot);
```

Comment:
add a todo remove these code when we ensure left child if from probe side

Reply:
done